.github/workflows/build_packages.yaml

```diff
@@ -372,12 +412,33 @@ jobs:
       with:
         username: ${{ secrets.DOCKER_HUB_USER }}
         password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+    - name: Get deps git refs for cache
+      id: deps-refs
+      run: |
+        cd source
+        make ensure-rebar3
+        sudo cp rebar3 /usr/local/bin/rebar3
+        scripts/get-dep-refs.sh
+        make clean-all
+
+    - name: load rocksdb cache
+      uses: actions/cache@v2
+      with:
+        path: source/_build/default/lib/rocksdb/
+        key: alpine3.14-${{ matrix.otp }}-${{ matrix.arch }}-${{ steps.deps-refs.outputs.DEP_ROCKSDB_REF }}
+    - name: load quicer cache
+      uses: actions/cache@v2
+      with:
+        path: source/_build/default/lib/quicer/
+        key: alpine3.14-${{ matrix.otp }}-${{ matrix.arch }}-${{ steps.deps-refs.outputs.DEP_QUICER_REF }}
+
     - uses: docker/build-push-action@v2
       with:
         push: ${{ github.event_name == 'release' && !github.event.release.prerelease }}
         pull: true
         no-cache: true
-        platforms: linux/amd64,linux/arm64
```

Comment:
not sure how this works.
this will appear to be two builds single-arch builds instead of one multi-arch build.
when pushing to docker registries.

does it mean the last 'push' action win?
or are docker registries smart enough to merge the pushes ?

Reply:
that was in the example of `docker/build-push-action@v2`
I just changed it to build matrix so that two builds could run in parallel and push to the docker registry accordingly.

Reply:
do you mean this ?
https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md

Reply:
You are right. Single-arch build overwritten each others.
 I made a PR to revert the change since the cache miss is high: https://github.com/emqx/emqx/pull/6738